ctx_test.go

```diff
@@ -2294,6 +2294,305 @@ func Benchmark_Ctx_Params(b *testing.B) {
+		require.Equal(t, uint8(8), Params[uint8](c, "number", 1))
+		require.Equal(t, uint8(8), Params[uint8](c, "number", 1, 2))
+		require.Equal(t, uint8(0), Params[uint8](c, "string"))
+		require.Equal(t, uint8(1), Params[uint8](c, "string", 1))
+		require.Equal(t, uint8(1), Params[uint8](c, "string", 1, 2))
+
+		// uint16
+		require.Equal(t, uint16(8), Params[uint16](c, "number"))
+		require.Equal(t, uint16(8), Params[uint16](c, "number", 1))
+		require.Equal(t, uint16(8), Params[uint16](c, "number", 1, 2))
+		require.Equal(t, uint16(0), Params[uint16](c, "string"))
+		require.Equal(t, uint16(1), Params[uint16](c, "string", 1))
+		require.Equal(t, uint16(1), Params[uint16](c, "string", 1, 2))
+
+		// uint32
+		require.Equal(t, uint32(8), Params[uint32](c, "number"))
+		require.Equal(t, uint32(8), Params[uint32](c, "number", 1))
+		require.Equal(t, uint32(8), Params[uint32](c, "number", 1, 2))
+		require.Equal(t, uint32(0), Params[uint32](c, "string"))
+		require.Equal(t, uint32(1), Params[uint32](c, "string", 1))
+		require.Equal(t, uint32(1), Params[uint32](c, "string", 1, 2))
+
+		// uint64
+		require.Equal(t, uint64(8), Params[uint64](c, "number"))
+		require.Equal(t, uint64(8), Params[uint64](c, "number", 1))
+		require.Equal(t, uint64(8), Params[uint64](c, "number", 1, 2))
+		require.Equal(t, uint64(0), Params[uint64](c, "string"))
+		require.Equal(t, uint64(1), Params[uint64](c, "string", 1))
+		require.Equal(t, uint64(1), Params[uint64](c, "string", 1, 2))
+
+		// string
+		require.Equal(t, "john", Params[string](c, "string"))
+		require.Equal(t, "john", Params[string](c, "string", "doe"))
+		require.Equal(t, "8", Params[string](c, "number"))
+		require.Equal(t, "", Params[string](c, "fake"))
+		require.Equal(t, "doe", Params[string](c, "fake", "doe"))
+		require.Equal(t, "doe", Params[string](c, "fake", "doe", "mary"))
+
+		// bool
+		require.True(t, Params[bool](c, "bool"))
+		require.True(t, Params[bool](c, "bool", false))
+		require.False(t, Params[bool](c, "string"))
+		require.True(t, Params[bool](c, "string", true))
+
+		// float32
+		require.InEpsilon(t, float32(3.1415), Params[float32](c, "float"), epsilon)
+		require.InEpsilon(t, float32(3.1415), Params[float32](c, "float", float32(1.234)), epsilon)
+		require.InDelta(t, float32(0), Params[float32](c, "string"), 0)
+		require.InEpsilon(t, float32(1.234), Params[float32](c, "string", float32(1.234)), epsilon)
+
+		// float64
+		require.InEpsilon(t, float64(3.1415), Params[float64](c, "float"), epsilon)
+		require.InEpsilon(t, float64(3.1415), Params[float64](c, "float", float64(1.234)), epsilon)
+		require.InDelta(t, float64(0), Params[float64](c, "string"), 0)
+		require.InEpsilon(t, float64(1.234), Params[float64](c, "string", float64(1.234)), epsilon)
+		return nil
+	})
+
+	resp, err := app.Test(httptest.NewRequest(MethodGet, "/test/john/8/true/3.1415", nil))
+	require.NoError(t, err, "app.Test(req)")
+	require.Equal(t, StatusOK, resp.StatusCode, "Status code")
+}
+
```

Comment:
```suggestion

    // go test -run Test_Params_TypeAssertFail
func Test_Params_TypeAssertFail(t *testing.T) {
	type customCtx struct {
		Ctx
	}

	app := New()
	c := &customCtx{
		Ctx: app.NewCtx(&fasthttp.RequestCtx{}),
	}

	defer func() {
		if r := recover(); r == nil {
			t.Errorf("The code did not panic")
		}
	}()

	Params[int](c, "key")
}

```

Reply:
and similar for `func Query[V GenericType](c Ctx, key string, defaultValue ...V) V {`